src/main/java/org/magicdgs/thaplv/utils/stats/LengthBinning.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package org.magicdgs.thaplv.utils.stats;
+
+import java.text.DecimalFormat;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+/**
+ * Class that bin a double based on the length, computing mean, median and quantiles.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+// TODO: implement unit tests
+public class LengthBinning {
+
+    // value that contains the bins
+    // TODO: implement a getter?
+    private final ConcurrentSkipListMap<Integer, RunningStats> binStat;
+
+    // value that contains the bin size
+    private final int binSize;
+
+    // which quantiles should be computed?
+    private final double[] quantiles;
+
+    private boolean empty;
+
+    /**
+     * Constructs a binning by length using the default quantiles (1, 5, 95, 99).
+     *
+     * @param binSize size of the bin.
+     */
+    public LengthBinning(final int binSize) {
+        // TODO: change contract to compute defaults with 0.005, 0.025, 0.975, 0.995 to have the 99% and 95%
+        this(binSize, 1, 5, 95, 99);
+    }
+
+    /**
+     * Constructs a binning by lenght using the provided quantiles.
+     *
+     * @param binSize   size of the bin.
+     * @param quantiles quantiles to add in the range (1, 100).
+     */
+    public LengthBinning(final int binSize, final double... quantiles) {
+        if (binSize < 1) {
+            throw new IllegalArgumentException(
+                    "Binning cannot be performed for bins lower than 1 bp");
+        }
+        this.binStat = new ConcurrentSkipListMap<>();
+        this.binSize = binSize;
+        this.quantiles = quantiles;
+        empty = true;
+    }
+
+    /** Returns {@code true} if no value was tracked in this object; {@code false} otherwise. */
+    public boolean isEmpty() {
+        return empty;
+    }
+
+    /**
+     * Add a value to the associated length bin.
+     *
+     * @param length the length associated with this value.
+     * @param value  the value to add to the computation of the statistic.
+     */
+    public void add(final int length, final double value) {
+        final int cBin = toBin(length);
+        RunningStats stats = binStat.get(cBin);
```

Comment:
The getter is synchronized, but the rest of the code is not. I think that the method should be synchronized and not the map itself.